src/Service/ServiceUtils.cpp

```diff
@@ -410,7 +410,7 @@ ErrorMessageOr<fs::path> FindSymbolsFilePath(
   return ErrorMessage("Unknown object file type.");
 }
 
-bool ReadProcessMemory(int32_t pid, uintptr_t address, void* buffer, uint64_t size,
+bool ReadProcessMemory(uint32_t pid, uintptr_t address, void* buffer, uint64_t size,
```

Comment:
this is linux-specific, use pid_t?

Reply:
It is but it's called through a GetProcessMemoryRequest which uses uint32_t. Added `CHECK(orbit_base::IsValidProcessId(pid));`

Reply:
Ok, I guess I am most concerned here that we do not have a clear boundary between native and interface level. I would explicitly split it into 2 components even - one linux only another one windows-only and have a proxy which will depending on compile flag call one or the other. This proxy would also be responsible for converting pids from interface type to platform specific one.

This is probably not for this PR, since this one is big as it is.

Reply:
Maybe add a TODO here to this regard so we do not forget.

Reply:
Yep agreed. This will come up very soon, with the PR for OrbitService on Windows.